### ISSUE TITLE
Mechanism Unit Tests

### ISSUE THREAD

**Issue by thorstenhater** [arbor-sim/arbor#1402]
Despite appearances, we have no actual unit tests on mechanisms, the file was disabled during a major re-work
(I assume) of the mechanism infrastructure and addition of catalogues. This has allowed at least one bug to sneak under the radar.

I propose to add a new test set that covers (on CPU and GPU)
- initialisation of storage
- attribute tables
- getting/setting attributes

The remaining functionality is part of the actual mechanism implemented by modcc and thus should be considered integration tests.

**Comment by thorstenhater** [comment]
This depends on #820 for consistency and foundation.

**Comment by schmitts** [comment]
Unit tests should cover, e.g., the implementation of a FUNCTION. The test infrastructure should be such that mechanisms and tests of parts of the mechanism can be easily added also for non-core developers, i.e., users that work on a custom mech.

**Comment by halfflat** [comment]
It's important that we verify the functional behaviour of mechanisms included with the Arbor library, and ensure that changes to e.g. modcc do not change the dynamical behaviour of these mechanisms. This could be a separate issue, or incorporated here.
